ext/sockets/sockets.c

```diff
@@ -1165,30 +1166,32 @@ PHP_FUNCTION(socket_getpeername)
    Creates an endpoint for communication in the domain specified by domain, of type specified by type */
 PHP_FUNCTION(socket_create)
 {
-	zend_long		arg1, arg2, arg3;
+	zend_long	domain, type, protocol;
 	php_socket	*php_sock = php_create_socket();
 
-	if (zend_parse_parameters(ZEND_NUM_ARGS(), "lll", &arg1, &arg2, &arg3) == FAILURE) {
+	if (zend_parse_parameters(ZEND_NUM_ARGS(), "lll", &domain, &type, &protocol) == FAILURE) {
 		efree(php_sock);
 		RETURN_THROWS();
 	}
 
-	if (arg1 != AF_UNIX
+	if (domain != AF_UNIX
 #if HAVE_IPV6
-		&& arg1 != AF_INET6
+		&& domain != AF_INET6
 #endif
-		&& arg1 != AF_INET) {
-		php_error_docref(NULL, E_WARNING, "Invalid socket domain [" ZEND_LONG_FMT "] specified for argument 1, assuming AF_INET", arg1);
-		arg1 = AF_INET;
+		&& domain != AF_INET) {
+		zend_argument_value_error(1, "is invalid, [" ZEND_LONG_FMT "] given", domain);
+		efree(php_sock);
+		RETURN_THROWS();
 	}
 
-	if (arg2 > 10) {
-		php_error_docref(NULL, E_WARNING, "Invalid socket type [" ZEND_LONG_FMT "] specified for argument 2, assuming SOCK_STREAM", arg2);
-		arg2 = SOCK_STREAM;
+	if (type > 10) {
+		zend_argument_value_error(2, "is invalid, [" ZEND_LONG_FMT "] given", type);
```

Comment:
It would be more helpful to list the accepted constants here as well (https://www.php.net/manual/en/function.socket-create.php)